tests/BeerAPI.Tests/Controllers/BeersControllersTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿using BeerAPI.Controllers;
+using BeerAPI.Models;
+using BeerAPI.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+
+namespace BeerAPI.Tests.Controllers
+{
+    public class BeersControllersTests
+    {
+        private readonly Mock<IBeerService> _mockBeerService;
+        private readonly Mock<IBeerDescriptionService> _mockBeerDescriptionService;
+        private readonly BeersController _controller;
+
+        public BeersControllersTests()
+        {
+            _mockBeerService = new Mock<IBeerService>();
+            _mockBeerDescriptionService = new Mock<IBeerDescriptionService>();
+            _controller = new BeersController(_mockBeerService.Object, _mockBeerDescriptionService.Object);
+        }
+
+
+        [Fact]
+        public void GetBeerById_NonExistingId_ReturnsNotFoundResult()
+        {
+            // Arrange
+            _mockBeerService.Setup(service => service.GetBeerById(It.IsAny<int>())).Returns(default(Beer?));
+
+            // Act
+            var result = _controller.GetBeerById(999); // this ID doesn't exist
+
+            // Assert
+            Assert.IsType<NotFoundResult>(result);
```

Comment:
Checking for `NotFoundResult` is a good start, but have a think about what that tells future engineers. Consider the  asserts below. What thoughts will a future engineer reading these different asserts have?
```suggestion
            var statusCodeResult = Assert.IsAssignableFrom<IStatusCodeActionResult>(result);
            Assert.Equal(404, statusCodeResult.StatusCode);
```

Reply:
makes sense!
`IsAssignableFrom` is less rigid than `IsType` (could be any type that implements the `IStatusCodeActionResult`) ad the checking of `statusCodeResult.StatusCode` being 404 means that we are telling future engineers what the status SHOULD be ! Thanks @ssu8 !